src/main/scala/play/dev/filewatch/DefaultFileWatchService.scala

```diff
@@ -0,0 +1,62 @@
+package play.dev.filewatch
+
+import java.io.File
+import java.nio.file._
+
+import better.files.{ File => ScalaFile, _ }
+import io.methvin.watcher.{ DirectoryChangeEvent, DirectoryChangeListener, DirectoryWatcher }
+import io.methvin.watchservice.MacOSXListeningWatchService
+
+import scala.collection.JavaConverters._
+import scala.util.control.NonFatal
+
+/**
+ * Implementation of the file watch service that uses a native implementation for Mac and otherwise uses the JDK's
+ * WatchService implementation.
+ */
+class DefaultFileWatchService(logger: LoggerProxy, isMac: Boolean) extends FileWatchService {
+
+  def this(logger: LoggerProxy) = this(logger, false)
+
+  def watch(filesToWatch: Seq[File], onChange: () => Unit) = {
+    val dirsToWatch = filesToWatch.filter { file =>
+      if (file.isDirectory) {
+        true
+      } else if (file.isFile) {
+        // JDK7 WatchService can't watch files
+        logger.warn("JDK7 WatchService only supports watching directories, but an attempt has been made to watch the file: " + file.getCanonicalPath)
+        logger.warn("This file will not be watched. Either remove the file from playMonitoredFiles, or configure a different WatchService, eg:")
+        logger.warn("PlayKeys.fileWatchService := play.runsupport.FileWatchService.jnotify(target.value)")
```

Comment:
This message needs to be updated. You are removing jnotify from `FileWatchService`, so it can't be recommended.

Reply:
Also, we will need to log this only when `watchService` is instantiated using `FileSystems.getDefault.newWatchService()`, right?

Reply:
Good point. I'll update just to say we don't support watching regular files. directory-watcher (as the name suggests) is only meant to watch directories.

Reply:
I suppose we could also split out jnotify into a separate dependency but since we do not really need the ability to watch directories in Play I'm not sure it's worth worrying about. Maybe for backwards compatibility it would be a good idea though.